kml/types.hpp

```diff
@@ -344,27 +344,39 @@ struct TrackLayer
 struct MultiGeometry
 {
   using LineT = std::vector<geometry::PointWithAltitude>;
+  using TimeT = std::vector<double>;
+
   std::vector<LineT> m_lines;
+  std::vector<TimeT> m_timestamps;
 
-  void Clear() { m_lines.clear(); }
-  bool IsValid() const { return !m_lines.empty(); }
+  void Clear();
+  bool IsValid() const;
 
   bool operator==(MultiGeometry const & rhs) const
   {
-    return IsEqual(m_lines, rhs.m_lines);
+    return IsEqual(m_lines, rhs.m_lines) && m_timestamps == rhs.m_timestamps;
   }
 
   friend std::string DebugPrint(MultiGeometry const & geom)
   {
-    return ::DebugPrint(geom.m_lines);
+    auto str = ::DebugPrint(geom.m_lines);
+    if (geom.HasTimestamps())
+      str += ::DebugPrint(geom.m_timestamps);
```

Comment:
Should some delimiter be added?
```suggestion
      str.append(" ").append(::DebugPrint(geom.m_timestamps));
```